gulp-tasks/scripts.js

```diff
@@ -0,0 +1,93 @@
+/*
+  Copyright 2016 Google Inc. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+/* eslint-env node */
+
+'use strict';
+
+const gulp = require('gulp');
+const uglify = require('gulp-uglify');
+const sourcemaps = require('gulp-sourcemaps');
+const browserify = require('browserify');
+const buffer = require('vinyl-buffer');
+const source = require('vinyl-source-stream');
+const path = require('path');
+const glob = require('glob');
+const eslint = require('gulp-eslint');
+
+const bundleJS = function(fullFilePath) {
+  const browserifyBundles = browserify({
+    entries: fullFilePath
+  });
+
+  // Rollupify reduces the size of the final output but increases build
+  // time to do it so enable for production build only
+  if (GLOBAL.config.env === 'prod') {
+    browserifyBundles.transform('rollupify');
+  }
+
+  let stream = browserifyBundles
+  .transform('babelify', {presets: ['es2015']})
+  .bundle()
+  // `source` Converts Browserify's Node Stream to a Gulp Stream
+  // Use path.relative to make the file have the correct home in `dest`
+  .pipe(
+    source(path.join('.', path.relative(GLOBAL.config.src, fullFilePath)))
+  )
+  .pipe(buffer())
+  .pipe(sourcemaps.init());
+
+  if (GLOBAL.config.env === 'prod') {
+    stream = stream.pipe(uglify());
+  }
+
+  return stream.pipe(sourcemaps.write('.'))
+  .pipe(gulp.dest(GLOBAL.config.dest));
+};
+
+const build = function() {
+  const globResponse = glob.sync(GLOBAL.config.src +
+    '/in-browser-helpers/**/*.js', {
+      dot: false
+    });
+
+  const buildPromise = globResponse.reduce((promise, filePath) => {
```

Comment:
Hmm, that's not what I'm seeing. It's a not big deal, but it's something that _should_ work, and appears to work when I've tested with the `Promise.all()` and multiple copies of the original script:

```sh
jeff@imac [/tmp/sw-testing-helpers]$ l src/in-browser-helpers/
-rw-r--r--  1 jeff  wheel  6631 Mar 30 20:09 sw-utils.js
-rw-r--r--  1 jeff  wheel  6631 Mar 30 20:10 sw-utils1.js
-rw-r--r--  1 jeff  wheel  6631 Mar 30 20:10 sw-utils2.js

jeff@imac [/tmp/sw-testing-helpers]$ gulp
[20:11:55] Using gulpfile /private/tmp/sw-testing-helpers/gulpfile.js
[20:11:55] Starting 'default'...
[20:11:55] Starting 'clean'...
[20:11:55] Finished 'clean' after 6.41 ms
[20:11:55] Starting 'scripts:lint'...
[20:11:56] Finished 'scripts:lint' after 730 ms
[20:11:56] Starting 'scripts'...
[20:11:56] Finished 'scripts' after 25 ms
[20:11:56] Finished 'default' after 764 ms

jeff@imac [/tmp/sw-testing-helpers]$ l build/in-browser-helpers/
-rw-r--r--  1 jeff  wheel   3186 Mar 30 20:11 sw-utils.js
-rw-r--r--  1 jeff  wheel  12121 Mar 30 20:11 sw-utils.js.map
-rw-r--r--  1 jeff  wheel   3187 Mar 30 20:11 sw-utils1.js
-rw-r--r--  1 jeff  wheel  12123 Mar 30 20:11 sw-utils1.js.map
-rw-r--r--  1 jeff  wheel   3187 Mar 30 20:11 sw-utils2.js
-rw-r--r--  1 jeff  wheel  12123 Mar 30 20:11 sw-utils2.js.map
```

Those generated files seem to be identical to what's generated when the Promise chain. What kind of output do you see when you run it with `Promise.all()`?